tests/code_style.py

```diff
@@ -0,0 +1,247 @@
+
+    spec = importlib.util.spec_from_file_location(module_path, path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+
+    return module
+
+
+def validate_test_names(transformer):
+    """Validate if the test methods are properly specified."""
+    test_file = _get_test_location(transformer)
+
+    # load the module
+    try:
+        module = _load_module_from_path(test_file)
+    except FileNotFoundError:
+        return False
+
+    # get test class
+    try:
+        test_class = getattr(module, f'Test{transformer.__name__}')
+
+    except AttributeError:
+        return False  # The test class doesn't exist
+
+    # get the test methods that start with test
+    test_functions = inspect.getmembers(test_class, predicate=inspect.isfunction)
+    test_functions = [
+        test
+        for test, _ in test_functions
+        if test.startswith('test')
+    ]
+
+    if not test_functions:
+        return False  # No test functions found
+
+    transformer_functions = [
+        name
+        for name, function in transformer.__dict__.items()
+        if isinstance(function, (FunctionType, classmethod, staticmethod))
+    ]
+
+    test_functions_is_valid = []
+
+    for test in test_functions:
+        count = len(test_functions_is_valid)
+        for transformer_function in transformer_functions:
+            simple_test = f'test_{transformer_function}'
+            described_test = f'test_{transformer_function}_'
+            if test.startswith(described_test):
+                test_functions_is_valid.append(True)
+            elif test.startswith(simple_test):
+                test_functions_is_valid.append(True)
+
+        if count == len(test_functions_is_valid):
+            test_functions_is_valid.append(False)
+
+    return all(test_functions_is_valid)
+
+
+def get_all_transformers():
+    """Return all the transformers."""
+    return BaseTransformer.get_subclasses()
+
+
+def get_all_transformers_as_dict():
+    """Return all the transformers."""
+    return {
+        transformer.__name__: transformer
+        for transformer in BaseTransformer.get_subclasses()
+    }
+
+
+def validate_transformer(transformer):
+    """Validate a transformer."""
+    if isinstance(transformer, str):
+        transformer = getattr(importlib.import_module('rdt.transformers'), transformer)
+
+    result = {
+        'transformer': transformer.__name__,
+        'valid_name': validate_transformer_name(transformer),
+        'is_subclass': validate_transformer_subclass(transformer),
+        'valid_transformer_module': validate_transformer_module(transformer),
+        'valid_test_location': validate_test_location(transformer),
+        'valid_test_naming': validate_test_names(transformer),
+    }
+
+    valid_addon = validate_transformer_addon(transformer)
+    if valid_addon is None:
+        valid_addon = 'NaN'
+
+    result['valid_addon'] = valid_addon
+    return result
+
+
+def validate_all_transformers():
```

Comment:
If we run this as a parameterized `test`, this and the `main` block below are not necessary any more.